packages/web3-utils/src/converters.ts

```diff
@@ -0,0 +1,340 @@
+import { BigDecimal } from './big_decimal';
+import { sha3 } from './hash';
+import { Address, Bytes, HexString, Numbers, ValueTypes } from './types';
+import { isAddress, isHexStrict } from './validation';
+
+const base = BigInt(10);
+const expo10 = (expo: number) => base ** BigInt(expo);
+
+// Ref: https://ethdocs.org/en/latest/ether.html
+/** @internal */
+const ethUnitMap = {
+	noether: BigInt('0'),
+	wei: BigInt(1),
+	kwei: expo10(3),
+	Kwei: expo10(3),
+	babbage: expo10(3),
+	femtoether: expo10(3),
+	mwei: expo10(6),
+	Mwei: expo10(6),
+	lovelace: expo10(6),
+	picoether: expo10(6),
+	gwei: expo10(9),
+	Gwei: expo10(9),
+	shannon: expo10(9),
+	nanoether: expo10(9),
+	nano: expo10(9),
+	szabo: expo10(12),
+	microether: expo10(12),
+	micro: expo10(12),
+	finney: expo10(15),
+	milliether: expo10(15),
+	milli: expo10(15),
+	ether: expo10(18),
+	kether: expo10(21),
+	grand: expo10(21),
+	mether: expo10(24),
+	gether: expo10(27),
+	tether: expo10(30),
+};
+
+const errorMessage = (value: unknown, error: string) =>
+	`Invalid value given "${String(value)}". Error: ${error}.`;
+
+export type EtherUnits = keyof typeof ethUnitMap;
+
+/** @internal */
+const bytesToBuffer = (data: Bytes): Buffer | never => {
+	if (Buffer.isBuffer(data)) {
+		return data;
+	}
+
+	if (data instanceof Uint8Array) {
+		return Buffer.from(data);
+	}
+
+	if (Array.isArray(data)) {
+		if (data.some(d => d < 0)) {
+			throw new Error(errorMessage(data, 'contains negative values'));
+		}
+
+		if (data.some(d => d > 255)) {
+			throw new Error(errorMessage(data, 'contains numbers greater than 255'));
+		}
+
+		if (data.some(d => !Number.isInteger(d))) {
+			throw new Error(errorMessage(data, 'contains invalid integer values'));
+		}
+
+		return Buffer.from(data);
+	}
+
+	if (typeof data === 'string' && !isHexStrict(data)) {
+		throw new Error(errorMessage(data, 'not valid hex string'));
+	}
+
+	if (typeof data === 'string' && (data.startsWith('0x') || data.startsWith('0X'))) {
+		return Buffer.from(data.substr(2), 'hex');
+	}
+
+	throw new Error(errorMessage(data, 'can not parse as byte data'));
+};
+
+/** @internal */
+const bufferToHexString = (data: Buffer) => `0x${data.toString('hex')}`;
+
+/**
+ * Convert a byte array to a hex string
+ */
+export const bytesToHex = (bytes: Bytes): HexString => bufferToHexString(bytesToBuffer(bytes));
+
+/**
+ * Convert a hex string to a byte array
+ */
+export const hexToBytes = (bytes: HexString): Buffer => bytesToBuffer(bytes);
+
+/**
+ * Converts value to it's number representation
+ */
+export const hexToNumber = (value: HexString): bigint | number => {
+	if (!isHexStrict(value)) {
+		throw new Error(errorMessage(value, 'not valid hex string'));
+	}
+
+	const [negative, hexValue] = value.startsWith('-') ? [true, value.substr(1)] : [false, value];
+	const num = BigInt(hexValue);
+
+	if (num > BigInt(Number.MAX_SAFE_INTEGER) || num < BigInt(Number.MIN_SAFE_INTEGER)) {
```

Comment:
This

```typescript
num < BigInt(Number.MIN_SAFE_INTEGER)
```

will never happen since the `-` is stripped off (via `value.substr(1)`) before checking